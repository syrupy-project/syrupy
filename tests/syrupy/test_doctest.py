def doctest_fn():
    """a doctest in a function docstring
    >>> doctest_fn() == getfixture('snapshot')
    True
    """
    return "doc test fn return value"


class DocTestClass:
    """
    >>> DocTestClass() == getfixture('snapshot')
    True

    a doctest in a class docstring
    >>> DocTestClass() == getfixture('snapshot')
    True
    """

    obj_attr = "test class attr"

    def doctest_method(self):
        """a doctest in a method docstring
        >>> DocTestClass().doctest_method() == getfixture('snapshot')
        True
        """
        return "doc test method return value"

    class NestedDocTestClass:
        """a doctest in a nested class docstring
        >>> DocTestClass.NestedDocTestClass() == getfixture('snapshot')
        True
        """

        nested_obj_attr = "nested doc test class attr"

        def doctest_method(self):
            """a doctest in a nested method docstring
            >>> nested_obj = DocTestClass.NestedDocTestClass()
            >>> nested_obj.doctest_method() == getfixture('snapshot')
            True
            """
            return "nested doc test method return value"
