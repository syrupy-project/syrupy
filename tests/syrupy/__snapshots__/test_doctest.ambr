# serializer version: 1
# name: DocTestClass
  DocTestClass(
    obj_attr='test class attr',
  )
# ---
# name: DocTestClass.NestedDocTestClass
  NestedDocTestClass(
    nested_obj_attr='nested doc test class attr',
  )
# ---
# name: DocTestClass.NestedDocTestClass.doctest_method
  'nested doc test method return value'
# ---
# name: DocTestClass.doctest_method
  'doc test method return value'
# ---
# name: DocTestClass.1
  DocTestClass(
    obj_attr='test class attr',
  )
# ---
# name: doctest_fn
  'doc test fn return value'
# ---
# name: test_doctest.txt
  'There must be a break after every snapshot assertion'
# ---
# name: test_doctest.txt.1
  'constant value'
# ---
# name: test_doctest.txt.2
  set({
    1,
    2,
    3,
  })
# ---
